KyberElasticAdapter/contracts/BaseAdapter.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "./interfaces/IERC20.sol";
+import "./interfaces/IWETH.sol";
+import "./lib/SafeERC20.sol";
+import "./lib/Maintainable.sol";
+
+abstract contract BaseAdapter is Maintainable {
+    using SafeERC20 for IERC20;
+
+    event BaseAdapterSwap(
```

Comment:
Missing NatSpec
Add descriptions for the interface, events and functions
Refer https://docs.soliditylang.org/en/v0.8.17/natspec-format.html